prov/verbs/src/verbs_msg.c

```diff
@@ -239,120 +239,8 @@ struct fi_ops_msg fi_ibv_msg_ep_msg_ops = {
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = (uintptr_t)msg->context,
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	if (flags & FI_REMOTE_CQ_DATA) {
-		wr.opcode = IBV_WR_SEND_WITH_IMM;
-		wr.imm_data = htonl((uint32_t)msg->data);
-	} else {
-		wr.opcode = IBV_WR_SEND;
-	}
-
-	return fi_ibv_send_msg(&ep->base_ep, &wr, msg, flags);
-}
-
-static ssize_t
-fi_ibv_msg_xrc_ep_send(struct fid_ep *ep_fid, const void *buf, size_t len,
-		void *desc, fi_addr_t dest_addr, void *context)
-{
-	struct fi_ibv_xrc_ep *ep = container_of(ep_fid, struct fi_ibv_xrc_ep,
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = VERBS_COMP(&ep->base_ep, (uintptr_t)context),
-		.opcode = IBV_WR_SEND,
-		.send_flags = VERBS_INJECT(&ep->base_ep, len),
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	return fi_ibv_send_buf(&ep->base_ep, &wr, buf, len, desc);
-}
-
-static ssize_t
-fi_ibv_msg_xrc_ep_senddata(struct fid_ep *ep_fid, const void *buf, size_t len,
-		       void *desc, uint64_t data, fi_addr_t dest_addr, void *context)
-{
-	struct fi_ibv_xrc_ep *ep = container_of(ep_fid, struct fi_ibv_xrc_ep,
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = VERBS_COMP(&ep->base_ep, (uintptr_t)context),
-		.opcode = IBV_WR_SEND_WITH_IMM,
-		.imm_data = htonl((uint32_t)data),
-		.send_flags = VERBS_INJECT(&ep->base_ep, len),
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	return fi_ibv_send_buf(&ep->base_ep, &wr, buf, len, desc);
-}
-
-static ssize_t
-fi_ibv_msg_xrc_ep_sendv(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
-		    size_t count, fi_addr_t dest_addr, void *context)
-{
-	struct fi_ibv_xrc_ep *ep = container_of(ep_fid, struct fi_ibv_xrc_ep,
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = (uintptr_t)context,
-		.opcode = IBV_WR_SEND,
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	return fi_ibv_send_iov(&ep->base_ep, &wr, iov, desc, count);
-}
-
-static ssize_t fi_ibv_msg_xrc_ep_inject(struct fid_ep *ep_fid, const void *buf, size_t len,
-		fi_addr_t dest_addr)
-{
-	struct fi_ibv_xrc_ep *ep = container_of(ep_fid, struct fi_ibv_xrc_ep,
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = VERBS_NO_COMP_FLAG,
-		.opcode = IBV_WR_SEND,
-		.send_flags = IBV_SEND_INLINE,
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	return fi_ibv_send_buf_inline(&ep->base_ep, &wr, buf, len);
-}
-
-static ssize_t fi_ibv_msg_xrc_ep_injectdata(struct fid_ep *ep_fid, const void *buf, size_t len,
-		    uint64_t data, fi_addr_t dest_addr)
-{
-	struct fi_ibv_xrc_ep *ep = container_of(ep_fid, struct fi_ibv_xrc_ep,
-						base_ep.util_ep.ep_fid);
-	struct ibv_send_wr wr = {
-		.wr_id = VERBS_NO_COMP_FLAG,
-		.opcode = IBV_WR_SEND_WITH_IMM,
-		.imm_data = htonl((uint32_t)data),
-		.send_flags = IBV_SEND_INLINE,
-		.qp_type.xrc.remote_srqn = ep->peer_srqn,
-	};
-
-	return fi_ibv_send_buf_inline(&ep->base_ep, &wr, buf, len);
-}
-
```

Comment:
Why are the above functions being removed?

Reply:
Thanks for finding this! It was removed by mistake.

Reply:
Fixed

Reply:
@jswaro @swelch Could take a look at the verbs/XRC modifications, please?

Reply:
@dmitrygladkov The XRC modifications look good to me.